rocAL/source/decoders/video/rocdec_video_decoder.cpp

```diff
@@ -0,0 +1,184 @@
+    int n_video_bytes = 0, n_frame_returned = 0, n_frame = 0, pkg_flags = 0;
+    OutputSurfaceInfo *surf_info;
+    uint8_t *pvideo = nullptr;
+    int num_decoded_frames = sequence_length * stride;
+
+    bool b_seek = true;       // seek only for first frame in sequence
+    uint32_t image_size = out_height * out_stride * sizeof(uint8_t);
+    video_seek_ctx.seek_crit_ = SEEK_CRITERIA_FRAME_NUM;
+    video_seek_ctx.seek_mode_ = SEEK_MODE_PREV_KEY_FRAME;
+    VideoPostProcess post_process;
+    bool sequence_decoded = false;
+    do {
+        if (b_seek) {
+            video_seek_ctx.seek_frame_ = static_cast<uint64_t>(seek_frame_number);
+            video_seek_ctx.seek_crit_ = SEEK_CRITERIA_FRAME_NUM;
+            video_seek_ctx.seek_mode_ = SEEK_MODE_PREV_KEY_FRAME;
+            _demuxer->Seek(video_seek_ctx, &pvideo, &n_video_bytes);
+            pts = video_seek_ctx.out_frame_pts_;
+            dts = video_seek_ctx.out_frame_dts_;
+            required_frame_dts = video_seek_ctx.selected_frame_dts_;
+            // std::cout << "info: Number of frames that were decoded during seek - " << video_seek_ctx.num_frames_decoded_ << std::endl;
+            b_seek = false;
+            _rocvid_decoder->FlushAndReconfigure();
+        } else {
+            _demuxer->Demux(&pvideo, &n_video_bytes, &pts, &dts);
+        }
+        // Treat 0 bitstream size as end of stream indicator
+        if (n_video_bytes == 0) {
+            pkg_flags |= ROCDEC_PKT_ENDOFSTREAM;
+        }
+        n_frame_returned = _rocvid_decoder->DecodeFrame(pvideo, n_video_bytes, pkg_flags, pts);
+        if (!n_frame && !_rocvid_decoder->GetOutputSurfaceInfo(&surf_info)) {
+            std::cerr << "Error: Failed to get Output Surface Info!" << std::endl;
+            break;
+        }
+        // Take the min of sequence length and num frames to avoid out of bounds memory error
+        int required_n_frames = std::min(static_cast<int>(sequence_length), n_frame_returned);
+        for (int i = 0; i < required_n_frames; i++) {
+            uint8_t *pframe = _rocvid_decoder->GetFrame(&pts);
+            if (dts >= required_frame_dts) {
+                if (n_frame % stride == 0) {
+                    post_process.ColorConvertYUV2RGB(pframe, surf_info, out_buffer, _output_format, _rocvid_decoder->GetStream());
+                    out_buffer += image_size;
+                }
+                n_frame++;
+            }
+            // release frame
+            _rocvid_decoder->ReleaseFrame(pts);
+            if (n_frame == num_decoded_frames) {
+                sequence_decoded = true;
+                break;
+            }
+        }
+
+        if (sequence_decoded) {
+            if (hipStreamSynchronize(_rocvid_decoder->GetStream()) != hipSuccess)
+                THROW("hipStreamSynchronize failed: ")
+            break;
+        }
+    } while (n_video_bytes);
+
+    return status;
+}
+
+void RocDecVideoDecoder::release() {
```

Comment:
No code in here, please recheck

Reply:
We are using shared pointers so all the resources are released